pkg/route/api/validation/validation.go

```diff
@@ -32,6 +32,11 @@ func ValidateRoute(route *routeapi.Route) fielderrors.ValidationErrorList {
 		result = append(result, fielderrors.NewFieldInvalid("path", route.Spec.Path, "path must begin with /"))
 	}
 
+	if len(route.Spec.Path) > 0 && route.Spec.TLS != nil &&
+		route.Spec.TLS.Termination == routeapi.TLSTerminationPassthrough {
```

Comment:
That argument could be made for v1 to v2 to v3 as well, no. My point is, if someone was using this - it would have never worked + the docs do mention it as well  
ref: https://docs.openshift.org/latest/architecture/core_concepts/routes.html#path-based-routes
We could argue the reverse direction to say hey someone could well do that but that's would cover a lot of area (the different planes of existence above) we don't know about.